SteamBot/Logging/Log.cs

```diff
@@ -95,10 +99,22 @@ public void RemoveLoggingObject(LoggerBase logger)
 
         public void Dispose()
         {
-            Disposed = true;
-            foreach (IDisposable logger in LoggerObjects.OfType<IDisposable>())
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        public void Dispose(bool disposing)
+        {
+            if (!Disposed)
             {
-                logger.Dispose();
+                if (disposing)
+                {
+                    foreach (IDisposable disposableLogger in LoggerObjects.OfType<IDisposable>())
+                        disposableLogger.Dispose();
```

Comment:
well, FileLogger has a destructor which internally calls a private method in FileLogger to close the file which from my understanding makes it a managed resource. The above code says if disposing is true, we should call Dispose on our managed resources, otherwise let garbagecollector call the destructor(aka the finalize method)